File: 1/js/2.js
const NEWLINE = "\r\n";

const fs = require("fs");

const remove = (x, arr) => arr.splice(arr.indexOf(x), 1);

(function () {
  const contents = fs.readFileSync("../input.txt").toString();
  const cals = contents.split(NEWLINE + NEWLINE).map(s => s.split(NEWLINE).map(x => +x).reduce((a, b) => a + b, 0));
  const max1 = Math.max(...cals);
  remove(max1, cals);
  const max2 = Math.max(...cals);
  remove(max2, cals);
  const max3 = Math.max(...cals);

  console.log(max1 + max2 + max3);
})();

File: 1/hs/2.hs
import System.IO
import Data.List
import Data.List.Split

main = do
  h <- openFile "../input.txt" ReadMode
  contents <- hGetContents h
  let
    cals = -- Array of calories per elf
      map sum $ -- Sum each array of calories
      (map.map) to_int $ -- Convert calorie strings to ints
      map lines $ -- Split strings into arrays of calorie strings
      splitOn "\n\n" contents -- Split into array of strings
      where
        to_int = read :: String -> Int
    max1 = maximum cals
    max2 = maximum $ delete max1 cals
    max3 = maximum $ delete max2 $ delete max1 cals
  print (max1 + max2 + max3)
  hClose h